source/slang/slang-parser.cpp

```diff
@@ -4747,6 +4747,88 @@ namespace Slang
 
         return modifier;
     }
+    static RefPtr<RefObject> parseSPIRVVersionModifier(Parser* parser, void* /*userData*/)
+    {
+        auto modifier = new RequiredSPIRVVersionModifier();
+
+        parser->ReadToken(TokenType::LParent);
+
+        UnownedStringSlice typeName;
+        UnownedStringSlice majorVersionText;
+        UnownedStringSlice minorVersionText;
+
+        // There is a legitimate question about how the version should be handled here.
+        switch (parser->tokenReader.peekTokenType())
+        {
+            case TokenType::FloatingPointLiteral:
+            {
+                modifier->token = parser->ReadToken(TokenType::FloatingPointLiteral);
+
+                List<UnownedStringSlice> split;
+                StringUtil::split(modifier->token.Content, '.', split);
+
+                if (split.getCount() != 2)
+                {
+                    parser->sink->diagnose(modifier->token, Diagnostics::invalidSPIRVVersion);
+                    return RefPtr<RefObject>();
+                }
+
+                majorVersionText = split[0];
+                minorVersionText = split[1];
+                break;
+            }
+            case TokenType::Identifier:
+            {
+                modifier->token = parser->ReadToken(TokenType::Identifier);
+
+                List<UnownedStringSlice> split;
+                StringUtil::split(modifier->token.Content, '_', split);
+
+                if (split.getCount() != 3)
+                {
+                    parser->sink->diagnose(modifier->token, Diagnostics::invalidSPIRVVersion);
+                    return RefPtr<RefObject>();
+                }
+
+                typeName = split[0];
+                majorVersionText = split[1];
+                minorVersionText = split[2];
+                break;
+
```

Comment:
The versioning was made overly flexible because of how it's handled internally in glslang. There is the convenient style 1.3, and the identifier style - which allows flexibility needed for say specifying vk version specific spir-v, as allowed for in glslang. 

That if there is a extra dot needed, you can long hand it to spirv_1_3_3_whatever_you_want_extra